modules/baremetal/main.tf

```diff
@@ -0,0 +1,36 @@
+resource "ibm_is_bare_metal_server" "bms" {
+  profile     = var.profile
+  name        = var.prefix
```

Comment:
`prefix` here is a bit misleading, we usually use the input `prefix` to supply a true "prefix" for a name, not the whole name.

If you intend for the input to be the whole name here, consider renaming it to something like `server_name` so that the user understands that the value will need to be unique and represent the true server name. Then below you can name your VNIs using the `server_name` with an index or something.